src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -109,49 +80,47 @@ internal ReturnTypeAndDiagnostics(TypeSymbol returnType, ImmutableArray<Diagnost
         /// </summary>
         internal Binder ScopeBinder { get; }
 
+        public Binder ParameterBinder => _binder;
+
         internal void GetDeclarationDiagnostics(DiagnosticBag addTo)
         {
-            // Force attribute binding for diagnostics
-            foreach (var typeParam in _typeParameters)
+            lock (_lazyLock)
             {
-                typeParam.GetAttributesBag(addTo);
-            }
+                // Force attribute binding for diagnostics
+                foreach (var typeParam in _typeParameters)
+                {
+                    typeParam.GetAttributesBag(_declarationDiagnostics);
```

Comment:
>typeParam.GetAttributesBag(_declarationDiagnostics) [](start = 20, length = 51)

It feels like we should simply call force complete on each type parameter and that should take care of adding diagnostics to the right bag ensuring that there would be no duplicates. The lock is not need then.